### ISSUE TITLE
Cannot install from tarball

### ISSUE THREAD

**Issue by Narvey** [npm/npm#20308]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [x] npm is producing incorrect or undesirable behavior.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
When I [ran into this error](https://github.com/bokuweb/re-resizable/issues/234), npm said "Note that you can also install from a tarball, folder, http url, or git url." so, since I didn't realize it was because of my old npm version, I wanted to try installing from the .zip I downloaded from their github. I looked in the [npm-debug.log](https://github.com/npm/npm/files/1899727/npm-debug.log), and it had the really retarded idea of nuking the extracted contents right after pulling them from the archive, so I upgraded npm from 3.10.8 to 5.6.0, and it still doesn't work, just in a different way. 

Now it gives `CwdError: ENOTDIR: Cannot cd into 'D:\Documents\form-tools-starter\node_modules\.staging\re-resizable-4f731185'` ([2018-04-11T16_48_50_687Z-debug.log](https://github.com/npm/npm/files/1899793/2018-04-11T16_48_50_687Z-debug.log))

#### How can the CLI team reproduce the problem?
1. Download https://github.com/bokuweb/re-resizable/archive/master.zip
2. Unzip, repackage (as either .tar or .zip: both break; different errors) so that package.json is in root of archive.
3. Go to any project folder and try to `npm install PATH/TO/re-resizeable-master.tar`

### supporting information:

 - `npm -v` prints: 5.6.0 (also tried with 3.10.8, see above)
 - `node -v` prints: v8.10.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Windows 10
 - Network issues:
   - Geographic location where npm was run: Colorado

**Comment by LakMoore** [comment]
I have the exact same problem.  Any ideas/solutions?
I'm using macOS and downloading the package as a tgz from my TeamCity CI server.

**Comment by LakMoore** [comment]
**TL;DR** npm might benefit from some mal-formed package checks!

I figured it out.  My tarball was malformed.  There is very little documentation out there on how to manually pack up an npm package.  Although there is an 'npm pack' command, which I would have used if I had noticed it sooner.  For anyone trying to make their own packages; everything must be contained within a root folder named 'package'!

package    <--   tarball this folder not just its contents
|-package.json
|-index.js

**Comment by legodude17** [comment]
@LakMoore This is what `npm pack` is for.
